app/Policies/AlbumPolicy.php

```diff
@@ -0,0 +1,277 @@
+	 * have any editable properties.
+	 * Hence, it is pointless whether a smart album is editable or not.
+	 * In order to silently ignore/skip this condition for smart albums,
+	 * this method always returns `true` for a smart album.
+	 *
+	 * @param User               $user
+	 * @param AbstractAlbum|null $album the album; `null` designates the root album
+	 *
+	 * @return bool
+	 */
+	public function canEdit(User $user, ?AbstractAlbum $album): bool
+	{
+		if (!$this->userPolicy->canUpload($user)) {
+			return false;
+		}
+
+		// The root album and smart albums get a pass
+		return
+			$album === null ||
+			$album instanceof BaseSmartAlbum ||
+			($album instanceof BaseAlbum && $this->isOwner($user, $album));
+	}
+
+	/**
+	 * Checks whether the album is visible by the current user.
+	 *
+	 * Note, at the moment this check is only needed for built-in smart
+	 * albums.
+	 * Hence, the method is only provided for them.
+	 *
+	 * @param User|null      $user
+	 * @param BaseSmartAlbum $smartAlbum
+	 *
+	 * @return bool true, if the album is visible
+	 */
+	public function isVisible(?User $user, BaseSmartAlbum $smartAlbum): bool
+	{
+		return ($user !== null && $this->userPolicy->canUpload($user)) ||
+			$smartAlbum->is_public;
+	}
+
+	/**
+	 * Checks whether the designated albums are editable by the current user.
+	 *
+	 * See {@link AlbumQueryPolicy::isEditable()} for the definition
+	 * when an album is editable.
+	 *
+	 * This method is mostly only useful during deletion of albums, when no
+	 * album models are loaded for efficiency reasons.
+	 * If an album model is required anyway (because it shall be edited),
+	 * then first load the album once and use
+	 * {@link AlbumQueryPolicy::isEditable()}
+	 * instead in order to avoid several DB requests.
+	 *
+	 * @param User  $user
+	 * @param array $albumIDs
+	 *
+	 * @return bool
+	 *
+	 * @throws QueryBuilderException
+	 */
+	public function canEditById(User $user, array $albumIDs): bool
+	{
+		if (!$this->userPolicy->canUpload($user)) {
+			return false;
+		}
+
+		// Remove root and smart albums, as they get a pass.
+		// Make IDs unique as otherwise count will fail.
+		$albumIDs = array_diff(
+			array_unique($albumIDs),
+			array_keys(AlbumFactory::BUILTIN_SMARTS),
+			[null]
+		);
+
+		return
+			count($albumIDs) === 0 ||
+			BaseAlbumImpl::query()
+			->whereIn('id', $albumIDs)
+			->where('owner_id', $user->id)
+			->count() === count($albumIDs);
+	}
+
+	// The following methods are not to be called by Gate.
+
+	/**
+	 * Pushes an album onto the stack of unlocked albums.
+	 *
+	 * @param BaseAlbum|BaseAlbumImpl $album
+	 */
+	public function unlock(BaseAlbum|BaseAlbumImpl $album): void
+	{
+		Session::push(AlbumPolicy::UNLOCKED_ALBUMS_SESSION_KEY, $album->id);
+	}
+
+	/**
+	 * Check whether the given album has previously been unlocked.
+	 *
+	 * @param BaseAlbum|BaseAlbumImpl $album
+	 *
+	 * @return bool
+	 */
+	public function unlocked(BaseAlbum|BaseAlbumImpl $album): bool
```

Comment:
I would prefer `isUnlocked` as it has been before. `unlocked` is no verb.

Reply:
Will fix.